documentation/docs/admin/production-setup/programming-exercise-adjustments.mdx

```diff
@@ -0,0 +1,392 @@
+It is extended by a `Jenkinsfile` in the same directory that will be placed inside the `config.xml` file.
+The `Jenkinsfile` handles the functionality shared by all programming languages like checking out the repositories and
+loading the actual exercise-specific pipeline script from the Artemis server.
+
+<Callout variant={CalloutVariant.info}>
+    When overriding the `Jenkinsfile` with a custom one, note that it **must** start either
+
+    - with `pipeline` (there must not be a comment before pipeline, but there can be one at any other position,
+      if the Jenkinsfile-syntax allows it)
+    - or the special comment `// ARTEMIS: JenkinsPipeline` in the first line.
+</Callout>
+
+The actual programming language or exercise-type specific pipeline steps are defined in the form of
+[scripted pipelines](https://www.jenkins.io/doc/book/pipeline/syntax/#scripted-pipeline).
+In principle, this is a Groovy script which allows structuring the pipeline into smaller methods and allows
+conditionally executing steps, but inside still allows the core structure blocks from
+[declarative pipelines](https://www.jenkins.io/doc/book/pipeline/syntax/#declarative-pipeline).
+You can override those `pipeline.groovy` files with the template mechanism described above.
+
+Inside the `pipeline.groovy` some placeholders exist that will be filled by Artemis upon exercise creation from the
+server or exercise settings:
+
+| Variable                       | Replacement                                                    | Origin                 |
+|--------------------------------|----------------------------------------------------------------|------------------------|
+| `#dockerImage`                 | The container image that the tests will run in.                | Server configuration   |
+| `#dockerArgs`                  | Additional flags passed to Docker when starting the container. | Server configuration   |
+| `#isStaticCodeAnalysisEnabled` | Defines if static code analysis should be performed.           | Exercise configuration |
+
+The `pipeline.groovy` file can be customized further by instructors after creating the exercise from within
+Artemis via the ‘Edit Build Plan’ button on the details page of the exercise.
+
+
+## Caching Maven Dependencies with Sonatype Nexus {#caching-sonatype-nexus}
+
+With Sonatype Nexus you can run a caching server in your local network for Maven dependencies.
+An alternative approach for caching is with Docker volumes, see [this section](#caching-docker-volumes).
+
+<Callout variant={CalloutVariant.info}>
+  The following steps assume `artemis.example.com` is the host `10.0.73.42` and is using port `8443` for the cache.
+  Adapt the URLs for your actual setup.
+</Callout>
+
+
+### Sonatype Nexus Setup
+
+1. Set up Sonatype Nexus to run on `artemis.example.com:8443`, e.g. in a [Docker container](https://hub.docker.com/r/sonatype/nexus3/>) behind a [proxy](https://help.sonatype.com/en/run-behind-a-reverse-proxy.html).
```

Comment:
The link is broken
```suggestion
1. Set up Sonatype Nexus to run on `artemis.example.com:8443`, e.g. in a [Docker container](https://hub.docker.com/r/sonatype/nexus3/) behind a [proxy](https://help.sonatype.com/en/run-behind-a-reverse-proxy.html).
```

Reply:
fixed: dc7736866b78cd60e4fd606c8780e6800206709e